<launch>
    <!-- Input arguments -->
    <arg name="simulated" default="false" />
    <arg name="carla" default="false" />
    <arg name="actor2" default="false" />


    <!-- Adding external ROS packages using SEPARATE LAUNCH FILES ............................. -->
    <!-- Makes it easy to update on the fly for each method -->
    <!-- These take priority in the launch order as actor_ros will pub/sub to these -->
    <group if="$(arg simulated)">
        <group if="$(arg carla)">
            <!-- TODO: Add support for CARLA here... -->
            <include file="$(find actor_ros)/launch/support_carla.launch" />
        </group>
        <group unless="$(arg carla)">
            <!-- Load the Simple Sim related launch file -->
            <include file="$(find actor_ros)/launch/support_sim.launch" />
        </group>
    </group>
    <group unless="$(arg simulated)">
        <!-- Load the Real vehicle related launch file -->
        <group if="$(arg actor2)">
            <include file="$(find actor_ros)/launch/support_real_actor_2.launch" />
        </group>
        <group unless="$(arg actor2)">
            <include file="$(find actor_ros)/launch/support_real_actor_1.launch" />
        </group>
    </group>


    <!-- Launch under the global namespace '/actor' ........................................... -->
    <group ns="/actor">

        <!-- Parameter used to determine simulation mode from other nodes -->
        <param name="is_simulated" value="$(arg simulated)" />
        <param name="is_carla" value="$(arg carla)" />

        <!-- Load Environment related parameters from YAML files -->
        <group if="$(arg simulated)">
            <group if="$(arg carla)">
                <!-- TODO: Add support for CARLA here... -->
                <rosparam command="load" file="$(find actor_ros)/cfg/topics_carla.yaml" />
            </group>
            <group unless="$(arg carla)">
                <!-- Load the Simple Sim related YAML file -->
                <rosparam command="load" file="$(find actor_ros)/cfg/topics_simulator.yaml" />
            </group>
        </group>
        <group unless="$(arg simulated)">
            <!-- Load the Real vehicle related YAML file -->
            <rosparam command="load" file="$(find actor_ros)/cfg/topics_real.yaml" />
        </group>
        <!-- .................................................................................. -->


        <!-- actor_ros specific nodes ......................................................... -->
        <!-- Actor Status -->
        <node name="actor_status" pkg="actor_ros" type="actor_status.py" respawn="true" />

        <!-- Actor Control -->
        <node name="actor_control" pkg="actor_ros" type="actor_control.py" respawn="true" />

        <!-- Actor GUI -->
        <node name="actor_gui" pkg="actor_ros" type="actor_gui.py" args="$(arg simulated)"
            respawn="true" output="screen" />
        <!-- Added simulated arg because gui does not run a ROS node to read ros params -->

        <!-- ROS Board -->
        <!-- <include file="$(find rosboard)/launch/rosboard.launch" /> -->
        <node name="rosboard" pkg="rosboard" type="rosboard_node" output="screen">
            <!-- If port is changed, please change it in GUI as well -->
            <param name="port" value="8888" />
        </node>
        <!-- .................................................................................. -->

    </group>
    <!-- ^^ End of /actor namespace ^^ ........................................................ -->

    <!-- NOTE: Do NOT add any other nodes here unless it is absolutely necessary -->
    <!-- Use the support_*.launch files to add support nodes for the respective environment -->

</launch>